File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-4.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        if(nums[0] >= target){
            return 1;
        }
        int l = 0;
        int res = 0;
        int sum = nums[0];

        for(int r = 1; r < nums.length; r++){
            sum += nums[r];
            while(sum >= target){
                if(Math.min(res, r-l+1) == 0){
                    res = r - l + 1;
                } else {
                    res = Math.min(res, r - l + 1);
                }
                if(r == l){
                    return 1;
                }
                l++;
                sum -= nums[l - 1];
            }
            System.out.print(r + " ");
            System.out.print(l + " ");
            System.out.println(res);
            
        }

        return res;
    }
}

File: Data Structures & Algorithms/duplicate-integer/submission-4.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        mapH = {}
        for num in nums:
            if mapH.get(num, 0) + 1 > 1:
                return True
            mapH[num] = mapH.get(num, 0) + 1
        return False     